core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java

```diff
@@ -699,7 +701,16 @@ else if (Tables.getTableState(context.getInstance(), table) == TableState.OFFLIN
 
     }
 
-    void addMutations(MutationSet mutationsToSend) {
+    void queueMutations(final MutationSet mutationsToSend) {
+      queueThreadPool.submit(new Runnable() {
+        public void run() {
+          addMutations(mutationsToSend);
```

Comment:
Need to ensure any exceptions are handled properly. If an expected exception happens here, mutations are lost, and the batch writer keeps working thats not good.